### PR TITLE
Spread _prepareContext_ middleware

### PR DESCRIPTION
**Pull request by dgaubert**
This PR includes:
- Removes _meta-middleware_ `prepareContext`.
- Reorder involved middlewares and use them explicitly for each controller
- Merge `allowQueryParams` and `cleanUpQueryParams` middlewares
- Fix ported tests by adding `host` header to indicate the user in the `req/res` cycle and remove the old `req2params` middleware.

### CODE REVIEW

File: lib/cartodb/controllers/named_maps.js [hunk 1]
```diff
@@ -222,9 +238,11 @@ function getTile ({ tileBackend, label }) {
                 return next(err);
             }
 
-            res.locals.body = tile;
-            res.locals.headers = headers;
-            res.locals.stats = stats;
```
Comment (oleurud) [inline lib/cartodb/controllers/named_maps.js]: don't we need to save the stats?

Comment (dgaubert) [reply lib/cartodb/controllers/named_maps.js]: We were doing `req.profiler.add(stats);` twice. Here and in `sendResponse` middleware.

Comment (oleurud) [reply lib/cartodb/controllers/named_maps.js]: Oki!

File: lib/cartodb/controllers/named_maps.js [hunk 2]
```diff
@@ -222,9 +238,11 @@ function getTile ({ tileBackend, label }) {
...
+            if (headers) {
+                res.set(headers);
+            }
+
+            res.body = tile;
```
Comment (oleurud) [inline lib/cartodb/controllers/named_maps.js]: :heart:

**Pull request merged**